twine/package.py

```diff
@@ -99,8 +100,10 @@ def from_filename(cls, filename: str, comment: Optional[str]) -> "PackageFile":
 
         py_version: Optional[str]
         if dtype == "bdist_egg":
-            pkgd = pkg_resources.Distribution.from_filename(filename)
-            py_version = pkgd.py_version
+            (dist,) = importlib_metadata.Distribution.discover(  # type: ignore
```

Comment:
The [mypy docs](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#when-you-re-puzzled-or-when-things-are-complicated) recommend indicating why `type: ignore` is used with an additional comment. I also discovered that one can now [ignore specific error codes](https://mypy.readthedocs.io/en/stable/error_codes.html#silencing-errors-based-on-error-codes). Could you add the error code here (and in the other `ignore`) for context?

Reply:
Alternatively, I wonder if `disallow_untyped_calls` could be disabled for `importlib_metadata` in Twine's mypy.ini. But maybe that's not how it works.

Reply:
Sure. Fixed in 91c2addf97c67c45914f96e53345e59c23bf172a and e27162f5e993c9b30644f31feecc7a4e1667d34d.